### PR TITLE
Adds release docs

### PR DESCRIPTION
**Pull request by periodic**
# Overview

Adds docs about our release process.

## Acceptance criteria

- The release process should be sufficiently described such that a novice engineer on the project can publish a new release.

## Testing plan

Went through the steps with v3.1.4

## Risks

None.

## Checklist

- [x] I added tests for this PR's change (or confirmed tests are not viable).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] I updated `Changelog.md` if this change is externally facing. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] I updated `*schema.json` if I have made changes for `.fossa.yml`, `fossa-deps.{json, yaml, yml}`. You may also need to update these if you have added/removed new dependency (e.g. pip) or analysis target type (e.g. poetry).
- [x] I linked this PR to any referenced GitHub issues, if they exist.

### CODE REVIEW

File: docs/contributing/releases.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Releases
+
+So you've made a change that you want to get out to users.  Great!  Let's get that released!
+
+We'll assume that you want to release all changes currently on `master`.  You should not release anything from another branch.
+
+1. First, make sure that you update the release notes.
+    1. Create a PR that updates the [Changelog.md](/Changelog.md).
```
Comment (meghfossa) [inline docs/contributing/releases.md:8]: `[Changelog.md](./../../Changelog.md)`

File: docs/contributing/releases.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    git checkout master
+    git pull origin
+
+    git tag -a vX.Y.Z -m "Releases vX.Y.Z"
+    git push origin vX.Y.Z
+    ```
+    Replace `vX.Y.Z` with your version number, such as `v3.1.4`.  The tag _must_ start with `v`.
+3. GitHub Actions will take the tag and run some tests before generating a draft release which can be found on the [releases page](https://github.com/fossas/fossa-cli/releases).
+4. If nothing has changed to dissuade you it's time to publish.
+    1. Add the notes for this release from [Changelog.md](/Changelog.md) to the release description.
```
Comment (meghfossa) [inline docs/contributing/releases.md:23]: `[Changelog.md](./../../Changelog.md)`

**Review by meghfossa — APPROVED** [review]
LGTM! 

you can run `npm install -g markdown-link-check
 && make check-links` to locally run link checking. In review comments, I have provided working reference that will work. I however, recommend getting markdown extension if you are in vscode, it will autocomplete path for you.

**Comment by periodic** [comment]
If I actually view the changelog it works fine.  I think this may be a case where Github's Markdown isn't standard markdown.  Do we want this to work well on github (goes from root of the project and so the link is more stable) or do we want it to be more portable (link is relative within your filesystem)?

**Comment by periodic** [comment]
Oh, I guess even Github recommends relative links.

> Relative links are easier for users who clone your repository. Absolute links may not work in clones of your repository - we recommend using relative links to refer to other files within your repository.

**Comment by periodic** [comment]
Huh, this time it failed because `cocoapods.org` is down.

**Pull request merged**